src/meta/duplication/meta_duplication_service.cpp

```diff
@@ -158,24 +173,34 @@ void meta_duplication_service::add_duplication(duplication_add_rpc rpc)
     const auto &request = rpc.request();
     auto &response = rpc.response();
 
-    LOG_INFO("add duplication for app({}), remote cluster name is {}",
-             request.app_name,
-             request.remote_cluster_name);
+    std::string remote_app_name;
+    if (request.__isset.remote_app_name) {
+        remote_app_name = request.remote_app_name;
+    } else {
+        // Once remote_app_name is not specified by client, use source app_name
+        // as remote_app_name to be compatible with old versions of client.
+        remote_app_name = request.app_name;
+    }
 
-    response.err = ERR_OK;
+    LOG_INFO("add duplication for app({}), remote cluster name is {}"
+             "remote app name is {}",
+             request.app_name,
+             request.remote_cluster_name,
+             remote_app_name);
 
     if (request.remote_cluster_name == get_current_cluster_name()) {
         response.err = ERR_INVALID_PARAMETERS;
-        response.__set_hint("illegal operation: adding duplication to itself");
-        return;
+        LOG_WARNING_DUP_HINT_AND_RETURN(response,
+                                        "illegal operation: adding duplication to itself");
     }
```

Comment:
How about define the macro as:
LOG_WARNING_DUP_HINT_AND_RETURN_IF_NOT(exp, resp, ...)

The error code is always `ERR_INVALID_PARAMETERS` so I think it can be ommited.